### PR TITLE
Update UPP control files for GFS, RRFS and HAFS

### LINKED ISSUE #643: Update the UPP control files used by GFS, RRFS and HAFS and upp/10.0.8
## Description
The UPP (EMC_post) develop branch was recently updated with g2tmp/1.10.0. The relevant control files for GFS, RRFS and HAFS were updated.  Also the upp/10.0.8 was released at the hpc-stack. It include some bug fixes in UPP code.

## Solution 

- Update the UPP control files of GFS, RRFS and HAFS for inline post
- Update the grib2 table from g2tmpl/1.10.0
- Update g2tmpl module to 1.10.0
- Update upp module to 10.0.8

## Alternatives
If applicable, add a description of any alternative solutions or features you've considered.

## Related to
Directly reference any issues or PRs in this or other repositories that this is related to, and describe how they are related. Example:
- required to support noaa-emc/nems/issues/<issue_number>
- needed by noaa-emc/fv3atm/pull/<pr_number>

### PR DESCRIPTION
**Pull request by WenMeng-NOAA**
# PR Checklist

- [x] Ths PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR. Please consult the ufs-weather-model [wiki](https://github.com/ufs-community/ufs-weather-model/wiki/Making-code-changes-in-the-UFS-weather-model-and-its-subcomponents) if you are unsure how to do this.

- [x] This PR has been tested using a branch which is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR

- [x] An Issue describing the work contained in this PR has been created either in the subcomponent(s) or in the ufs-weather-model. The Issue should be created in the repository that is most relevant to the changes in contained in the PR. The Issue and the dependent sub-component PR 
are specified below.

- [ ] If new or updated input data is required by this PR, it is clearly stated in the text of the PR.

## Instructions: All subsequent sections of text should be filled in as appropriate.

The information provided below allows the code managers to understand the changes relevant to this PR, whether those changes are in the ufs-weather-model repository or in a subcomponent repository. Ufs-weather-model code managers will use the information provided to add any applicable labels, assign reviewers and place it in the Commit Queue. Once the PR is in the Commit Queue, it is the PR owner's responsiblity to keep the PR up-to-date with the develop branch of ufs-weather-model. 

## Description

Provide a detailed description of what this PR does. What bug does it fix, or what feature does it add? Is a change of answers expected from this PR? Are any library updates included in this PR (modulefiles etc.)?
This PR is for

- Updating the post control files for GFS, RRFS an
- Updating grib2 table params_grib2_tbl_new used in inline post
- Updating upp lib to 10.0.8
- Updating g2tmpl lib to 1.10.0

There will be changes of grib2 files from inline post for GFS, RRFS and HAFS.

### Issue(s) addressed

Link the issues to be closed with this PR, whether in this repository, or in another repository.
(Remember, issues must always be created before starting work on a PR branch!) 
- fixes #643

## Testing

How were these changes tested? What compilers / HPCs was it tested with? Are the changes covered by regression tests? (If not, why? Do new tests need to be added?) Have regression tests and unit tests (utests) been run? On which platforms and with which compilers? (Note that unit tests can only be run on tier-1 platforms)

- [ ] hera.intel
- [x] hera.gnu
- [ ] orion.intel
- [x] cheyenne.intel 
- [x] cheyenne.gnu
- [x] gaea.intel 
- [x] jet.intel
- [x] wcoss_cray
- [x] wcoss_dell_p3
- [x] CI

## Dependencies

None

### CODE REVIEW

File: modulefiles/ufs_common [hunk 1]
```diff
@@ -13,10 +13,10 @@ module load fms/2020.04.03
 module load bacio/2.4.1
 module load crtm/2.3.0
 module load g2/3.4.1
```
Comment (junwang-noaa) [inline modulefiles/ufs_common]: According to Kyle, we need g2/3.4.2

Comment (WenMeng-NOAA) [reply modulefiles/ufs_common]: In my offline and inline post testing, g2/3.4.1 was used. Let me test g2/3.4.2 in UPP first.

**Review by climbfuji — APPROVED** [review]
I don't know much about the postconfig files, the rest looks good to me.

**Review by junwang-noaa — APPROVED** [review]

**Review by DusanJovic-NOAA — APPROVED** [review]

**Comment by BrianCurtis-NOAA** [comment]
Machine: hera
Compiler: intel
Job: BL
Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/670620714/20210617014520/ufs-weather-model
Please make changes and add the following label back:
hera-intel-BL

**Comment by BrianCurtis-NOAA** [comment]
Machine: jet
Compiler: intel
Job: BL
Repo location: /lfs4/HFIP/h-nems/emc.nemspara/autort/pr/670620714/20210617013010/ufs-weather-model
Please manually delete: /lfs4/HFIP/h-nems/emc.nemspara/RT_RUNDIRS/emc.nemspara/FV3_RT/rt_269986
Baseline creation and move successful
Repo location: /lfs4/HFIP/h-nems/emc.nemspara/autort/pr/670620714/20210617031749/ufs-weather-model
Please manually delete: /lfs4/HFIP/h-nems/emc.nemspara/RT_RUNDIRS/emc.nemspara/FV3_RT/rt_96577
Test control_c384 021 failed failed 
Test control_c384 021 failed in run_test failed 
Please make changes and add the following label back:
jet-intel-BL

**Comment by junwang-noaa** [comment]
on jet:
[186:x159] unexpected DAPL connection event 0x4008 from 376
Fatal error in PMPI_Wait: Internal MPI error!, error stack:
PMPI_Wait(219)...........: MPI_Wait(request=0x1e365ba0, status=0x1) failed
MPIR_Wait_impl(81).......: fail failed
PMPIDI_CH3I_Progress(850): fail failed
(unknown)(): Internal MPI error!

The control_c384 test was rerun and passed.

**Comment by BrianCurtis-NOAA** [comment]
Machine: hera
Compiler: intel
Job: BL
Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/670620714/20210617123010/ufs-weather-model
Please manually delete: /scratch1/NCEPDEV/stmp2/emc.nemspara/FV3_RT/rt_129165
Please make changes and add the following label back:
hera-intel-BL

**Comment by junwang-noaa** [comment]
> 
> 
> Machine: hera
> Compiler: intel
> Job: BL
> Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/670620714/20210617123010/ufs-weather-model
> Please manually delete: /scratch1/NCEPDEV/stmp2/emc.nemspara/FV3_RT/rt_129165
> Please make changes and add the following label back:
> hera-intel-BL

@BrianCurtis-NOAA hera-intel-BL failed again, can you check?

**Comment by junwang-noaa** [comment]
@MinsukJi-NOAA In this PR, the g2(3.4.2), g2tmpl (1.10.0) and upp (10.0.8) libraries are updated, do we need to update the CI test to load those new libraries?

**Comment by MinsukJi-NOAA** [comment]
> @MinsukJi-NOAA In this PR, the g2(3.4.2), g2tmpl (1.10.0) and upp (10.0.8) libraries are updated, do we need to update the CI test to load those new libraries?

The CI tests passed, so I think it is okay for this PR. But I am going to update and test the container with those libraries for a future PR.

**Comment by BrianCurtis-NOAA** [comment]
```
[18:17:49]Brian.Curtis@hfe08:/scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/670620714/20210617123010/ufs-weather-model/tests
-->tail -f RegressionTests_hera.intel.log
 Moving RESTART/sfc_data.tile6.nc .........OK

  0: The total amount of wall time                        = 332.127790

Test 080 control_atmwav PASS


REGRESSION TEST WAS SUCCESSFUL
Thu Jun 17 13:09:20 UTC 2021
Elapsed time: 00h:36m:50s. Have a nice day!
```
Seems like it was OK, all files were moved over to the baseline storage and it doesn't look like anything is missing. I've added the RT label, hopefully it goes quick, apologies for the delay.

**Comment by junwang-noaa** [comment]
@BrianCurtis-NOAA I manually submit hera intel RT, it's now done.

@WenMeng-NOAA the hera intel log file is available at:/scratch1/NCEPDEV/nems/Jun.Wang/nems/vlab/20210616/test/ufs-weather-model/tests/RegressionTests_hera.intel.log
Please commit it to your branch

**Comment by BrianCurtis-NOAA** [comment]
Looks like the hera.intel label is done, just waiting on the comparison and log file upload.

**Comment by BrianCurtis-NOAA** [comment]
@WenMeng-NOAA looks like the AutoRT finished and uploaded a successful log file for you. You can skip uploading it yourself.

**Comment by WenMeng-NOAA** [comment]
@junwang-noaa , @BrianCurtis-NOAA Thanks for running rt tests and committing the log files.

**Pull request merged**